### PR TITLE
Products & Coproducts

### LINKED ISSUE #7: 1.5-products-and-coproducts.md

### PR DESCRIPTION
**Pull request by raulraja**
Fixes #7

### CODE REVIEW

File: src/main/ank/1.5-products-and-coproducts.md [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+```kotlin
+fun <A> absurd(): (Nothing) -> A
+```
+................
+```Haskell
+unit :: a -> ()
+unit _ = ()
+```
+```kotlin:ank:silent
+val unit: Unit = Unit
```
Comment (nomisRev) [inline src/main/ank/1.5-products-and-coproducts.md]: This doesn't have the same signature in Kotlin as in Haskell, does it?

File: src/main/ank/1.5-products-and-coproducts.md [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+```
+................
+```Haskell
+yes :: a -> Bool
+yes _ = True
+```
+```kotlin:ank:silent
+val yes: Boolean = true
```
Comment (nomisRev) [inline src/main/ank/1.5-products-and-coproducts.md]: Same

File: src/main/ank/1.5-products-and-coproducts.md [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+```
+................
+```Haskell
+no :: a -> Bool
+no _ = False
+```
+```kotlin:ank:silent
+val no: Boolean = false
```
Comment (nomisRev) [inline src/main/ank/1.5-products-and-coproducts.md]: Same

Comment (raulraja) [reply src/main/ank/1.5-products-and-coproducts.md]: I can make them fun if it's more clear

Comment (raulraja) [reply src/main/ank/1.5-products-and-coproducts.md]: I can make them fun if it's more clear

Comment (raulraja) [reply src/main/ank/1.5-products-and-coproducts.md]: @nomisRev is this closer to what you envisioned or do you mean using `val` with `(A) -> B` return types?

File: src/main/ank/1.5-products-and-coproducts.md (hunk 1 above)
Comment (LordRaydenMK) [reply src/main/ank/1.5-products-and-coproducts.md]: for Types and Functions I used `val unit: () -> Unit = { }` - not sure if it's correct

File: src/main/ank/1.5-products-and-coproducts.md (hunk 3 above)
Comment (nomisRev) [reply src/main/ank/1.5-products-and-coproducts.md]: Exactly what I was thinking of.

File: src/main/ank/1.5-products-and-coproducts.md (hunk 1 above)
Comment (nomisRev) [reply src/main/ank/1.5-products-and-coproducts.md]: @LordRaydenMK `val` cannot be used with generics unless it specifies a generic receiver.

**Review by nomisRev — APPROVED** [review]

**Pull request merged**